lib/passport-saml/q-shim.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * A collection of Q-flavored helper functions implemented with native promises
+ */
+
+/* global Promise */
+
+/**
+ * Calls a function and casts the returned value to a promise
+ * @param {function():any} fn
+ * @returns {Promise<any>}
+ */
+function fcall(fn) {
+  try {
+    var result = fn();
+    return Promise.resolve(result);
+  } catch (err) {
+    return Promise.reject(err);
+  }
+}
```

Comment:
`fcall` seems like it could replaced with marking a function as `async` now, right?